### PR TITLE
Location limits

### PR DESCRIPTION
**Pull request by neoancient**
This is the third and final set of changes that validates equipment construction rules, mostly for vehicles. The first checked against motive type, the second against mount location, and this one takes care of the more tedious work of checking the maximum number in a location. For mechs this also streamlines and completes the checks for required or prohibited actuators.

Fixes MegaMek/megameklab#500

### CODE REVIEW

File: megamek/src/megamek/common/verifier/TestEntity.java [hunk 1]
```diff
@@ -1354,6 +1354,19 @@ public boolean hasIllegalEquipmentCombinations(StringBuffer buff) {
...
                 robotics++;
             }
+            if (m.getType().hasFlag(MiscType.F_LIFTHOIST)) {
+                liftHoists++;
+            } else if ((m.getLocation() > 0)
+                    && ((m.getType().hasFlag(MiscType.F_CLUB) && !((MiscType) m.getType()).isShield())
+                    || m.getType().hasFlag(MiscType.F_BULLDOZER)
+                    || m.getType().hasFlag(MiscType.F_HAND_WEAPON))) {
+                physicalWeaponsByLocation.putIfAbsent(m.getLocation(), new ArrayList<>());
+                physicalWeaponsByLocation.get(m.getLocation()).add(m.getType());
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/verifier/TestEntity.java]: Any objection to?
```suggestion
                physicalWeaponsByLocation.computeIfAbsent(m.getLocation(), new ArrayList<>()).add(m.getType());
```

File: megamek/src/megamek/common/verifier/TestEntity.java [hunk 2]
```diff
@@ -1419,7 +1432,27 @@ public boolean hasIllegalEquipmentCombinations(StringBuffer buff) {
             buff.append("Stealth armor requires an ECM generator.\n");
             illegal = true;
         }
-        
+        if ((getEntity() instanceof Mech) && (liftHoists > 2)) {
+            illegal = true;
+            buff.append("Can mount a maximum of two lift hoists.\n");
+        } else if ((getEntity().isSupportVehicle() || (getEntity() instanceof Tank)) &&  (liftHoists > 4)) {
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/verifier/TestEntity.java]:
```suggestion
        } else if ((getEntity().isSupportVehicle() || (getEntity() instanceof Tank)) && (liftHoists > 4)) {
```

File: megamek/src/megamek/common/verifier/TestEntity.java (hunk 1 above)
Comment (neoancient) [reply megamek/src/megamek/common/verifier/TestEntity.java]: The second parameter of computeIfAbsent is a Function, so it needs to be
```
physicalWeaponsByLocation.computeIfAbsent(m.getLocation(), ArrayList::new).add(m.getType());

Comment (sixlettervariables) [reply megamek/src/megamek/common/verifier/TestEntity.java]: Apologies, attempted the suggestion from memory.

**Pull request merged**